rust/chains/abacus-ethereum/src/validator_manager.rs

```diff
@@ -92,11 +93,19 @@ where
     M: Middleware + 'static,
 {
     #[tracing::instrument(err, skip(self))]
-    async fn submit_checkpoint(
+    async fn process(
         &self,
         multisig_signed_checkpoint: &MultisigSignedCheckpoint,
+        message: &AbacusMessage,
+        proof: &Proof,
     ) -> Result<TxOutcome, ChainCommunicationError> {
-        let tx = self.contract.cache_checkpoint(
+        let mut sol_proof: [[u8; 32]; 32] = Default::default();
+        sol_proof
```

Comment:
nit:
```rs
for i in 0..32 {
  sol_proof[i] = proof.path[i].to_fixed_bytes();
}
```

I am not certain if this compiles to the same thing or not but I just think it is a little cleaner and _might_ get unrolled this way and not the other.

Reply:
This made clippy upset :(
```
warning: the loop variable `i` is used to index `sol_proof`
   --> chains/abacus-ethereum/src/validator_manager.rs:103:18
    |
103 |         for i in 0..32 {
    |                  ^^^^^
    |
    = note: `#[warn(clippy::needless_range_loop)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
help: consider using an iterator
    |
103 |         for (i, <item>) in sol_proof.iter_mut().enumerate() {
    |             ~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Reply:
I think I could manually disable the clippy warning if you think that's better?

Reply:
I think I am acting too much like a C++ dev. Go ahead and keep it as it was.